packages/firestore/src/local/indexeddb_migrations.ts

```diff
@@ -0,0 +1,123 @@
+/**
+ * Copyright 2018 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import {
+  DbDocumentMutation,
+  DbInstance,
+  DbMutationBatch,
+  DbMutationQueue,
+  DbOwner,
+  DbRemoteDocument,
+  DbTarget,
+  DbTargetDocument,
+  DbTargetGlobal
+} from './indexeddb_schema';
+import { assert, fail } from '../util/assert';
+
+// TODO(mikelehen): Get rid of "as any" if/when TypeScript fixes their types.
+// https://github.com/Microsoft/TypeScript/issues/14322
+type KeyPath = any; // tslint:disable-line:no-any
+
+function createCache(db: IDBDatabase): void {
```

Comment:
FWIW, "cache" is non-obvious, so I would at least add a comment.

// Create IndexedDB stores for our query / remote document caches.

Reply:
I renamed this and split this up to match the names in Android:  createQueryCache() and createRemoteDocumentCache()